vault/auth.go

```diff
@@ -457,6 +461,9 @@ func (c *Core) teardownCredentials() error {
 // newCredentialBackend is used to create and configure a new credential backend by name
 func (c *Core) newCredentialBackend(
 	t string, sysView logical.SystemView, view logical.Storage, conf map[string]string) (logical.Backend, error) {
+	if alias, ok := credentialAliases[t]; ok {
```

Comment:
@joelthompson Yes, that's what I'm suggesting. There can be scripts that are expecting a certain value when checking whether mounts are correctly setup (in fact we have one such script internally). I don't really see a reason to break this and our general M.O. when possible is "keep old behavior, but only document the new". So new users will use `aws`, old users have a choice as to whether to keep using `aws-ec2` or switch over.

This code is only really necessary if we remove the cli capability to add the backend as `aws-ec2` which I think we should retain (but not document). What this code doesn't do is change the entry's `type` which is probably the right way to go.